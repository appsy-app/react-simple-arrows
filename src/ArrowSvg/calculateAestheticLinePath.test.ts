import { LineOrientation } from '../constants';
import { calculateAestheticLinePath } from './calculateAestheticLinePath';

describe('calculateAestheticLinePath', () => {
  const directionCombinations = [
    {
      name: 'up and right',
      start: { x: 0, y: 0 },
      end: { x: 500, y: 200 },
    },
    {
      name: 'down and right',
      start: { x: 500, y: 0 },
      end: { x: 0, y: 200 },
    },
    {
      name: 'up and left',
      start: { x: 0, y: 200 },
      end: { x: 500, y: 0 },
    },
    {
      name: 'down and left',
      start: { x: 500, y: 200 },
      end: { x: 0, y: 0 },
    },
  ];
  describe('vertical', () => {
    directionCombinations.forEach((directionCombination) => {
      it(`should return the expected path for line going ${directionCombination.name}`, () => {
        const path = calculateAestheticLinePath({
          start: directionCombination.start,
          end: directionCombination.end,
          curviness: 0.5,
          orientation: LineOrientation.VERTICAL,
        });
        expect(path).not.toContain('-'); // should not have negatives
        expect(path).toMatchSnapshot();
      });
    });
  });
  describe('horizontal', () => {
    directionCombinations.forEach((directionCombination) => {
      it(`should return the expected path for line going ${directionCombination.name}`, () => {
        const path = calculateAestheticLinePath({
          start: directionCombination.start,
          end: directionCombination.end,
          curviness: 0.5,
          orientation: LineOrientation.HORIZONTAL,
        });
        expect(path).not.toContain('-'); // should not have negatives
        expect(path).toMatchSnapshot();
      });
    });
  });
});
