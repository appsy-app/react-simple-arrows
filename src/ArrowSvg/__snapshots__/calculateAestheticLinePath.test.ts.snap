// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateAestheticLinePath horizontal should return the expected path for line going down and left 1`] = `"M 500 200 C 250 200 250 0 0 0"`;

exports[`calculateAestheticLinePath horizontal should return the expected path for line going down and right 1`] = `"M 500 0 C 250 0 250 200 0 200"`;

exports[`calculateAestheticLinePath horizontal should return the expected path for line going up and left 1`] = `"M 0 200 C 250 200 250 0 500 0"`;

exports[`calculateAestheticLinePath horizontal should return the expected path for line going up and right 1`] = `"M 0 0 C 250 0 250 200 500 200"`;

exports[`calculateAestheticLinePath vertical should return the expected path for line going down and left 1`] = `"M 500 200 C 500 100 0 100 0 0"`;

exports[`calculateAestheticLinePath vertical should return the expected path for line going down and right 1`] = `"M 500 0 C 500 100 0 100 0 200"`;

exports[`calculateAestheticLinePath vertical should return the expected path for line going up and left 1`] = `"M 0 200 C 0 100 500 100 500 0"`;

exports[`calculateAestheticLinePath vertical should return the expected path for line going up and right 1`] = `"M 0 0 C 0 100 500 100 500 200"`;
